realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -115,6 +115,7 @@ public DynamicRealmObject(RealmModel obj) {
      */
     public boolean getBoolean(String fieldName) {
         long columnIndex = proxyState.getRow$realm().getColumnIndex(fieldName);
+        checkFieldType(fieldName, columnIndex, RealmFieldType.BOOLEAN);
```

Comment:
The `UncheckedRow` does not check. And since `DynamicRealmObject` also has `get(String fieldName)`, we use `UncheckedRow`.